src/core/event.js

```diff
@@ -482,9 +482,19 @@ Candy.Core.Event = (function(self, Strophe, $) {
 			}
 
 			if(directInvite.length > 0) {
+				fromUser = Candy.Core.getRoster().get(msg.attr('from'));
```

Comment:
So, open question (including @bklang): how do we treat users who aren't in our roster? Do we pass through the JID instead of a Contact, or do we pass something API-compatible with Contact but which nevertheless indicates absence of full details from the roster and is scoped only to this event?